src/main/java/de/tum/in/www1/artemis/domain/quiz/DropLocationCounter.java

```diff
@@ -1,31 +1,25 @@
 package de.tum.in.www1.artemis.domain.quiz;
 
-import jakarta.persistence.CascadeType;
-import jakarta.persistence.DiscriminatorValue;
-import jakarta.persistence.Entity;
-import jakarta.persistence.JoinColumn;
-import jakarta.persistence.ManyToOne;
-import jakarta.persistence.OneToOne;
-
 import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonInclude;
 
+import de.tum.in.www1.artemis.domain.TempIdObject;
+
 /**
  * A DropLocationCounter.
  */
-@Entity
-@DiscriminatorValue(value = "DD")
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
-public class DropLocationCounter extends QuizStatisticCounter implements QuizQuestionStatisticComponent<DragAndDropQuestionStatistic, DropLocation, DragAndDropQuestion> {
+public class DropLocationCounter extends TempIdObject implements QuizQuestionStatisticComponent<DragAndDropQuestionStatistic, DropLocation, DragAndDropQuestion> {
 
-    @ManyToOne
     @JsonIgnore
     private DragAndDropQuestionStatistic dragAndDropQuestionStatistic;
 
-    @OneToOne(cascade = { CascadeType.PERSIST })
-    @JoinColumn(unique = true)
     private DropLocation dropLocation;
 
+    private Integer ratedCounter = 0;
+
+    private Integer unratedCounter = 0;
```

Comment:
This is now duplicated for every former subclass of `QuizStatisticCounter`. Why is this necessary?